jcl/src/java.base/share/classes/jdk/internal/misc/Unsafe.java

```diff
@@ -6668,7 +6813,9 @@ private static char convEndian(boolean isBigEndian, char value) {
 	 * @param value primitive type to store in obj
 	 */
 	public final <V> void putValueVolatile(Object obj, long offset, Class<?> clz, V v) {
-		putValue(obj, offset, clz, v);
+		synchronized (inlineTypesLock) {
+			putValue(obj, offset, clz, v);
+		}
 	}
 
```

Comment:
You can also add implementations for the following get/put methods. They should all be wrappers to existing ones that you have already implemented.

Reply:
added in 571fade